pkg/operator/nodekubeconfigcontroller/nodekubeconfigcontroller.go

```diff
@@ -0,0 +1,181 @@
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configv1informers "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	configv1listers "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/v410_00_assets"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+const workQueueKey = "key"
+
+type NodeKubeconfigController struct {
+	operatorClient v1helpers.StaticPodOperatorClient
+
+	kubeClient          kubernetes.Interface
+	configMapLister     corev1listers.ConfigMapLister
+	secretLister        corev1listers.SecretLister
+	infrastuctureLister configv1listers.InfrastructureLister
+}
+
+func NewNodeKubeconfigController(
+	operatorClient v1helpers.StaticPodOperatorClient,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	kubeClient kubernetes.Interface,
+	infrastuctureInformer configv1informers.InfrastructureInformer,
+	eventRecorder events.Recorder,
+) factory.Controller {
+	c := &NodeKubeconfigController{
+		operatorClient:      operatorClient,
+		kubeClient:          kubeClient,
+		configMapLister:     kubeInformersForNamespaces.ConfigMapLister(),
+		secretLister:        kubeInformersForNamespaces.SecretLister(),
+		infrastuctureLister: infrastuctureInformer.Lister(),
+	}
+
+	return factory.New().WithInformers(
+		operatorClient.Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.OperatorNamespace).Core().V1().ConfigMaps().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().ConfigMaps().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.OperatorNamespace).Core().V1().Secrets().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().Secrets().Informer(),
+		infrastuctureInformer.Informer(),
+	).WithSync(c.sync).ResyncEvery(time.Second).ToController("NodeKubeconfigController", eventRecorder.WithComponentSuffix("node-kubeconfig-controller"))
+}
+
+func (c NodeKubeconfigController) sync(ctx context.Context, syncContext factory.SyncContext) error {
+	operatorSpec, _, _, err := c.operatorClient.GetStaticPodOperatorState()
+	if err != nil {
+		return err
+	}
+
+	switch operatorSpec.ManagementState {
+	case operatorv1.Managed:
+	case operatorv1.Unmanaged:
+		return nil
+	case operatorv1.Removed:
+		// TODO probably just fail
```

Comment:
nit: Should this be addressed pre- or post-merge?

Reply:
it lasted for 6 releases in the other controller, this is just copy paste